### ISSUE TITLE
PPO 训练中回复切片与掩码边界在左填充场景下不一致

### ISSUE THREAD

**Issue by Azengjin** [jingyaogong/minimind#533]
博主，您好
我有一个问题。

<img width="1137" height="483" alt="Image" src="https://github.com/user-attachments/assets/1748b632-19fa-4022-8158-667332fccf32" />

代码在左侧填充（ padding=True ）的批次中，使用 prompt_lengths = enc.attention_mask.sum(dim=1) 作为“回复段起点”来切片与构造掩码，会把提示尾部的真实 token 误计入“回复段”。
正确的“回复段起点”应是批内统一的输入宽度 input_width = enc.input_ids.shape[1] （左填充后的宽度），而不是每个样本的非 pad 长度。
我的建议是：将原本的 prompt_lengths = enc.attention_mask.sum(dim=1) 修改为 prompt_lengths = torch.full((enc.input_ids.size(0),), enc.input_ids.shape[1], dtype=torch.long, device=enc.input_ids.device)

**Comment by jingyaogong** [comment]
正确的，已更新😊
